packages/@webex/plugin-cc/test/unit/spec/AgentConfig/AgentConfig.ts

```diff
@@ -0,0 +1,276 @@
+    expect(getDesktopProfileByIdSpy).toHaveBeenCalledOnceWith;
+    expect(getListOfTeamsSpy).toHaveBeenCalledOnceWith;
+    expect(getListOfAuxCodesSpy).toHaveBeenCalledOnceWith;
+
+    const result = await agentConfig.getAgentProfile();
+    expect(result).toEqual(expectedProfile);
+  });
+
+  it('should handle errors when fetching agent profile', async () => {
+    getUserUsingCISpy.mockRejectedValue(new Error('Network error'));
+
+    await expect(agentConfig.getAgentProfile()).rejects.toThrow(
+      'Fetching Agent Profile failed: Network error'
+    );
+  });
+
+  it('should handle errors when fetching desktop profile', async () => {
+    getDesktopProfileByIdSpy.mockRejectedValue(new Error('API error'));
+
+    await expect(agentConfig.getAgentProfile()).rejects.toThrow(
+      'Fetching Agent Profile failed: API error'
+    );
+  });
+
+  it('should handle errors when fetching list of teams', async () => {
+    getListOfTeamsSpy.mockRejectedValue(new Error('API error'));
+
+    await expect(agentConfig.getAgentProfile()).rejects.toThrow(
+      'Fetching Agent Profile failed: API error'
+    );
+  });
+
+  it('should handle errors when fetching list of aux codes', async () => {
+    getListOfAuxCodesSpy.mockRejectedValue(new Error('API error'));
+
+    await expect(agentConfig.getAgentProfile()).rejects.toThrow(
+      'Fetching Agent Profile failed: API error'
+    );
+  });
+
+  it('should handle both accessIdleCode and accessWrapUpCode as ALL', async () => {
+    const allDesktopProfileResponse: DesktopProfileResponse = {
+      loginVoiceOptions: ['option1', 'option2'],
+      accessWrapUpCode: 'ALL',
+      accessIdleCode: 'ALL',
+      wrapUpCodes: [],
+      idleCodes: [],
+    };
+
+    getDesktopProfileByIdSpy.mockResolvedValue(allDesktopProfileResponse);
+
+    const allAuxCodesResponse: ListAuxCodesResponse = {
+      data: [
+        { id: 'aux1', active: true, defaultCode: true, isSystemCode: true, description: 'test', name: 'testName', workTypeCode: WORK_TYPE_CODE.WRAP_UP_CODE},
+        { id: 'aux1', active: true, defaultCode: true, isSystemCode: true, description: 'test', name: 'testName', workTypeCode: WORK_TYPE_CODE.IDLE_CODE}
+      ],
+    };
+
+    getListOfAuxCodesSpy.mockResolvedValue(allAuxCodesResponse);
+
+    const expectedProfile: IAgentProfile = {
+      agentId: 'agent123',
+      name: 'John Doe',
+      agentProfileId: 'profile123',
+      agentMailId: 'john.doe@example.com',
+      teams: mockTeamsListResponse,
+      loginVoiceOptions: ['option1', 'option2'],
+      wrapUpCodes: [{id: 'aux1', active: true, defaultCode: true, isSystemCode: true, description: 'test', name: 'testName', workTypeCode: WORK_TYPE_CODE.WRAP_UP_CODE}],
+      idleCodes: [{id: 'aux1', active: true, defaultCode: true, isSystemCode: true, description: 'test', name: 'testName', workTypeCode: WORK_TYPE_CODE.IDLE_CODE}]
+    };
+
+    expect(getUserUsingCISpy).toHaveBeenCalledOnceWith;
```

Comment:
Fix these expect statements as well. Fix in all the testcases here

Reply:
Discussed over Webex & made changes according to this.